dart/lib/src/protocol/mechanism.dart

```diff
@@ -42,27 +42,27 @@ class Mechanism {
   /// An optional flag indicating that this error is synthetic.
   /// Synthetic errors are errors that carry little meaning by themselves.
   /// This may be because they are created at a central place (like a crash handler), and are all called the same: Error, Segfault etc. When the flag is set, Sentry will then try to use other information (top in-app frame function) rather than exception type and value in the UI for the primary event display. This flag should be set for all "segfaults" for instance as every single error group would look very similar otherwise.
-  final bool synthetic;
+  final bool? synthetic;
 
   Mechanism({
-    @required this.type,
+    required this.type,
     this.description,
     this.helpLink,
     this.handled,
     this.synthetic,
-    Map<String, dynamic> meta,
-    Map<String, dynamic> data,
-  })  : _meta = meta != null ? Map.from(meta) : null,
-        _data = data != null ? Map.from(data) : null;
+    Map<String, dynamic>? meta,
+    Map<String, dynamic>? data,
+  })  : _meta = Map.from(meta ?? {}),
+        _data = Map.from(data ?? {});
```

Comment:
the downside of doing this is allocating a Map when its not necessary, if values are nullable, we dont need to do it right

Reply:
Yeah, that's why I added comments about the `const` lints. That way it is not a real performance penalty and non-nullable lists are nicer to work with.
See also https://stackoverflow.com/a/50431087/12002462

Reply:
great, thanks